enterprise/security/src/test/java/org/neo4j/server/security/enterprise/auth/BuiltInProceduresInteractionTestBase.java

```diff
@@ -638,17 +640,38 @@ public void shouldHaveSetTXMetaDataProcedure() throws Throwable
     }
 
     @Test
-    public void shouldFailOnTooLargeTXMetaData() throws Throwable
+    public void readUpdatedMetadataValue() throws Exception
     {
-        ArrayList<String> list = new ArrayList<>();
-        for ( int i = 0; i < 200; i++ )
+        GraphDatabaseFacade localGraph = neo.getLocalGraph();
+        try ( Transaction transaction = localGraph.beginTx() )
         {
-            list.add( format( "key%d: '0123456789'", i ) );
+            String testValue = "testValue";
+            String testKey = "test";
+            localGraph.execute( "CALL dbms.setTXMetaData({" + testKey + ":'" + testValue + "'})" );
+            Map<String,Object> metadata =
+                    (Map<String,Object>) localGraph.execute( "CALL dbms.getTXMetaData " ).next().get( "metadata" );
+            assertEquals( testValue, metadata.get( testKey ) );
+        }
+    }
+
+    @Test
+    public void readEmptyMetadataInOtherTransaction()
+    {
+        String testValue = "testValue";
+        String testKey = "test";
+        GraphDatabaseFacade localGraph = neo.getLocalGraph();
```

Comment:
You should not need to get the localGraph or create separate transactions here. Perhaps:

    assertEmpty( writeSubject, "CALL dbms.setTXMetaData( { "+testKey+":"+testValue+" } )" );
    assertSuccess( writeSubject, "CALL dbms.getTXMetaData()", r -> {...} );